client/landing/stepper/declarative-flow/internals/hooks/use-step-navigation-with-tracking/can-use-automatic-go-back/index.tsx

```diff
@@ -0,0 +1,16 @@
+const parseUrl = ( url: string ) => {
+	try {
+		return new URL( url );
+	} catch ( error ) {
+		return null;
+	}
+};
+
+export const canUseAutomaticGoBack = () => {
+	const referrer = parseUrl( document.referrer );
+	if ( ! referrer ) {
+		return false;
+	}
+
+	return referrer.host === window.location.host;
+};
```

Comment:
Sadly this doesn't work. :( 

It turns out the referrer value is not updated on `history.push` so if you visit /setup/onboarding as a first destination and navigate to the plans step, the `document.referrer` will remain empty and the back button won't appear.

We need another check in addition to this one.

Reply:
This week I couldn't continue the work due other priorities but begin next week I will finish I it.